tests/operator_test.go

```diff
@@ -2480,26 +2482,46 @@ spec:
 			// This label shouldn't exist, but this isn't harmful
 			// existing/running deployments will not be torn down until
 			// new ones are stood up (and the new ones will get stuck in scheduling)
+			var pods *k8sv1.PodList
 			labelKey := "kubevirt-test"
 			labelValue := "test-label"
 			infra := v1.ComponentConfig{
 				NodePlacement: &v1.NodePlacement{
 					NodeSelector: map[string]string{labelKey: labelValue},
 				},
 			}
+			By("Adding fake label to Virt components")
 			patchKvInfra(&infra, false, "")
-
-			Eventually(func() bool {
-				for _, name := range []string{"virt-controller", "virt-api"} {
+			for _, name := range []string{"virt-controller", "virt-api"} {
+				Eventually(func() error {
 					deployment, err := virtClient.AppsV1().Deployments(flags.KubeVirtInstallNamespace).Get(context.Background(), name, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
 					if deployment.Spec.Template.Spec.NodeSelector == nil || deployment.Spec.Template.Spec.NodeSelector[labelKey] != labelValue {
-						return false
+						return fmt.Errorf("NodeSelector should be propegated to the deployment eventually")
 					}
-				}
-				return true
-			}, 60*time.Second, 1*time.Second).Should(BeTrue())
+					return nil
+				}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
 
+				Eventually(func() error {
```

Comment:
Don't you want to move this after `patchKvInfra(nil, false, "")` ?

Reply:
The `patchKvInfra(nil, false, "")` will cause the pending pod to be deleted and then the `Eventually` will fail if they will be deleted before the Eventually scope starts.

Reply:
But won't the `patchKvInfra` below trigger another update that can be missed by the `allPodsAreReady`?

Reply:
All it does is restoring the original kubevirtCR to the way it was